mlflow/server/handlers.py

```diff
@@ -262,6 +262,43 @@ def wrapper(*args, **kwargs):
 ]
 
 
+def _disable_unless_serving_artifacts(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        from mlflow.server import SERVE_ARTIFACTS_ENV_VAR
+
+        if not os.environ.get(SERVE_ARTIFACTS_ENV_VAR):
+            return Response(
+                (
+                    "Endpoints for mlflow artifacts service are disabled. To enable them, "
+                    "run `mlflow server` with `--serve-artfiacts-opt`"
+                ),
+                503,
+            )
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def _disable_mlflow_artifacts_only(func):
```

Comment:
```suggestion
def _disable_if_artifacts_only(func):
```

Can we also rename this function?

Reply:
much more clear decorator name. changed

Reply:
@BenWilson2 It appears it's not changed yet :)